junit-runtime/src/main/scala/org/scalajs/junit/JUnitTask.scala

```diff
@@ -17,29 +17,38 @@ final class JUnitTask(val taskDef: TaskDef, runner: JUnitBaseRunner)
   }
 
   def execute(eventHandler: EventHandler, loggers: Array[Logger]): Array[Task] = {
-    val richLogger = new RichLogger(loggers, runner.runSettings,
-        taskDef.fullyQualifiedName)
+    val fullClassName = taskDef.fullyQualifiedName
+    val richLogger = new RichLogger(loggers, runner.runSettings, fullClassName)
 
     if (runner.runSettings.verbose)
       richLogger.info(c("Test run started", INFO))
 
-    val bootstrapperName = taskDef.fullyQualifiedName + "$scalajs$junit$bootstrapper"
+    val bootstrapperName = fullClassName + "$scalajs$junit$bootstrapper"
 
     val startTime = System.nanoTime
 
+    def errorWhileLoadingClass(msg: String): Unit = {
+      val msg = {
+        "Error while loading test class: " + fullClassName + "\n" +
+        " caused by:" + msg
+      }
+      richLogger.error(msg)
+      val selector = new TestSelector(fullClassName)
+      eventHandler.handle(new JUnitEvent(taskDef, Status.Failure, selector))
+    }
+
     Try(TestUtils.loadModule(bootstrapperName, runner.testClassLoader)) match {
       case Success(classMetadata: JUnitTestBootstrapper) =>
         new JUnitExecuteTest(taskDef, runner, classMetadata,
             richLogger, eventHandler).executeTests()
 
       case Success(_) =>
-        richLogger.error("Error while loading test class: " +
-            taskDef.fullyQualifiedName + ", expected " + bootstrapperName +
-            " to extend JUnitTestBootstrapper")
+        errorWhileLoadingClass(
+            s"expected $bootstrapperName to extend JUnitTestBootstrapper")
```

Comment:
Consider also reporting the class of what you got. If this case ever occurs, it is probably because of a name clash. In that case reporting the class might be useful.